pkg/operator/controller/externaldns/deployment.go

```diff
@@ -441,6 +471,69 @@ func externalDNSVolumesChanged(current, expected, updated *appsv1.Deployment) bo
 	return changed
 }
 
+// externalDNSDeploymentStrategyChanged returns true if the current update strategy differs from expected.
+func externalDNSDeploymentStrategyChanged(current, expected, updated *appsv1.Deployment) bool {
+	changed := false
+
+	// reset the strategy type if it differs
+	if expected.Spec.Strategy.Type != "" &&
+		expected.Spec.Strategy.Type != current.Spec.Strategy.Type {
+		updated.Spec.Strategy.Type = expected.Spec.Strategy.Type
+		changed = true
+	}
+
+	// if the expected strategy is Recreate: make sure no rolling update params are present
+	if expected.Spec.Strategy.Type == appsv1.RecreateDeploymentStrategyType {
+		if current.Spec.Strategy.RollingUpdate != nil {
+			updated.Spec.Strategy.RollingUpdate = nil
+			changed = true
+		}
+	}
+
+	// if the expected strategy is Rolling: set the expected parameters if any
+	if expected.Spec.Strategy.Type == appsv1.RollingUpdateDeploymentStrategyType &&
```

Comment:
I thought this PR was to implement the enforcement of the recreate strategy.  Why is the rolling update strategy allowed?

Reply:
Yeah, that was the dilemma I had.
At first I implemented `externalDNSDeploymentStrategyChanged` as a strict enforcer of  the`Recreate` strategy only. Then I looked at the name of the function and I saw that it doesn't really do what it says. I got blown away by this idea. Also because I knew that [the supporting of the multiple replicas](https://github.com/kubernetes-sigs/external-dns/issues/2430) is around upstream.